tmt/base.py

```diff
@@ -838,7 +838,24 @@ def __init__(
     def _expand_node_data(self, data):
         """ Recursively expand variables in node data """
         if isinstance(data, str):
-            return os.path.expandvars(data)
+            # expand env variables
+            data_expanded_vars = os.path.expandvars(data)
+            # convert context values to string and define env variables using context
+            # keys and values
+            list_context_string = [
+                (a,
+                 ','.join(b)) if isinstance(
+                    b,
+                    list) else (
+                    a,
+                    str(b)) for (
+                    a,
+                    b) in self._fmf_context().items()]
```

Comment:
Hmm, could this be refactored for improved readability? It's hard to follow `a` and `b`, and the indentation is also quite unexpected. Something like this, maybe?

```python
list_context_string = [
    (key, ','.join(value) if isinstance(value, list) else str(value))
    for key, value in self._fmf_context().items()
    ]
```

Reply:
I can change variable names but the indentation is according to what autopep8 asked me to do. See
https://github.com/teemtee/tmt/actions/runs/3245488693/jobs/5323087268#step:4:126

Reply:
Changed

Reply:
Understood, variable names would help a lot. Can't do much about the autopep8.

---

(mumbling angrily) Autopep8 should be punished as much as possible for this abomination. It helps absolutely no one, the line length is 99 characters, there's no reason to spread a two-liner across so many lines... It's actually hurting people's ability to understand the code :/